apps/webapp/utils/calculate-balance.ts

```diff
@@ -0,0 +1,9 @@
+import { Asset } from 'penumbra-constants';
+import { LoHi } from 'penumbra-types';
+
+export const calculateBalance = (loHi: LoHi, asset: Asset): number => {
+  // found exponent from denomUnits by display name
+  const exponent = asset.denomUnits.find(denom => denom.denom === asset.display)?.exponent;
+
+  return (Number(loHi.lo) + 2 ** 64 * Number(loHi.hi)) / (exponent ? 10 ** exponent : 1);
+};
```

Comment:
In `packages/types/src/lo-hi.ts`, we have `joinLoHi`. Let's use that. If that doesn't cut it, let's just add core logic that is missing there. Putting it in a package let's other apps/packages make use of it.

Reply:
Yes, please transfer that to the hi/lo package and write tests for it. Also, please include documentation for what that formula is doing. Also, don't do `Number(loHi.lo)` and coerce it to a number in the first step. Do all the math as BigInt and at the very end you can coerce to a number. If you do it too early, you'll lose precision.